src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(HashMap storage, FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        if (quantity < 0) {
+            throw new RuntimeException(String.format("Error setting balance "
+                    + "of the fruit %s: quantity %d is less than zero", fruit, quantity));
+        }
+        storage.put(fruit, quantity);
+    }
```

Comment:
The `BalanceOperation` class should not have direct access to the `storage`. According to the SOLID principles, particularly the Single Responsibility Principle, this class should only be responsible for the logic of the balance operation. The storage access should be encapsulated within a separate class or service.